### ISSUE TITLE
`initialPage` option is really usable only for the first set of Viewer tiled images

### ISSUE THREAD

**Issue by VipSaran** [openseadragon/openseadragon#1191]
Initializing the Viewer in `sequenceMode: true` with an `initialPage: x` (where `x` is a positive int > 0) makes sense only for the first set of tiled images that is `open`ed. This is because the `initialPage` value is retained in the Viewer member variable, but the tiled images are "always" provided new, and quite possibly, with different `tileSources` lengths, hence making all non-zero `initialPage` values "out of context".

The problem with this that I stumbled across is the following:
* sequenceMode: true
* initialPage: 9
* Viewer.open(tileSources), tileSources.length = 10
* Viewer.goToPage(9)
* Viewer loads 1 image (last one) --> OK
* Viewer.open(new tileSources), tileSources.length = 3
* Viewer.goToPage(1)
* Viewer loads 2 images (last one b/c initialPage is out of range and then the go-to one) --> NOK

What I'm proposing is this signature (and some pre-processing) for the [Viewer.open()](https://openseadragon.github.io/docs/openseadragon.js.html#line7295) function:
```
    /**
     * Open tiled images into the viewer, closing any others.
     * @function
     * @param {Array|String|Object|Function} tileSources - This can be a TiledImage
     * specifier, a TileSource specifier, or an array of either. A TiledImage specifier
     * is the same as the options parameter for {@link OpenSeadragon.Viewer#addTiledImage},
     * except for the index property; images are added in sequence.
     * A TileSource specifier is anything you could pass as the tileSource property
     * of the options parameter for {@link OpenSeadragon.Viewer#addTiledImage}.
     * @param {Number} initialPage - If sequenceMode is true, display this page initially
     * for the given tileSources. If specified, will override tileSources option.
     * @return {OpenSeadragon.Viewer} Chainable.
     * @fires OpenSeadragon.Viewer.event:open
     * @fires OpenSeadragon.Viewer.event:open-failed
     */
    open: function (tileSources, initialPage) {
        var _this = this;

        this.close();

        if (!tileSources) {
            return;
        }

        if (this.sequenceMode && $.isArray(tileSources)) {
            if (this.referenceStrip) {
                this.referenceStrip.destroy();
                this.referenceStrip = null;
            }

            var _initialPage = this.initialPage;
            if (typeof initialPage != 'undefined' && !isNaN(initialPage)) {
              _initialPage = initialPage;
            }

            this.tileSources = tileSources;
            this._sequenceIndex = Math.max(0, Math.min(this.tileSources.length - 1, _initialPage));
```

This simple extension retains backward compatibility and eliminates loading of unwanted images.

Sounds sensible? :)

**Comment by iangilman** [comment]
Good catch! 

Another way to deal with this given the current API would be to set `viewer.initialPage = 1` before calling `viewer.open` the second time. Your proposed API extension basically makes this option more explicit, which seems reasonable. I think it should probably set the `viewer.initialPage` to the new value as well. 

Would you be up for making a patch for this?

**Comment by VipSaran** [comment]
Well, somehow I thought option variables (like `viewer.initialPage`) are "private" (i.e. not accessible/changeable during runtime), as they (possibly) shouldn't be (b/c changing them in-flight could cause unexpected results). Hence I did not try that approach -- which should also work :)

I'm not going to pretend to see the big picture or understand all implications regarding the Viewer options handling, so I'll provide a patch based on what you suggest should happen with `viewer.initialPage` value after the new `viewer.open()` call. Even though I'd treat an(y) option as immutable, supplying it as a parameter (in my suggested extension) does seem to imply that the "global" `initialPage` is also affected.

So, what I want to say, I'm fine with either, just let me know your (final) preference.

**Comment by iangilman** [comment]
Yeah, unfortunately some of the initial options are changeable after the fact and others are not and we haven't done a very good job of documenting which is which. The ones that are changeable should probably be documented separately as members of the Viewer object. In the case of `initialPage` of course it only affects subsequent calls to `open`. 

So yeah, I think my suggestion stands, but you might also add a JSDoc comment for `initialPage` as a Viewer member while you're in there. That should go somewhere around: 

https://github.com/openseadragon/openseadragon/blob/master/src/viewer.js#L92

...and be similar to the member comment for `element` a few lines below. 

Thank you for doing this!

**Comment by iangilman** [comment]
Fixed by #1196
